pyriemann/utils/covariance.py

```diff
@@ -28,90 +85,103 @@ def _fpcm(X, *, init=None, tol=10e-3, n_iter_max=50, assume_centered=False):
         If `True`, data will not be centered before computation.
         Useful when working with data whose mean is almost, but not exactly
         zero. If `False`, data will be centered before computation.
+    q : float, default=0.5
+        Using Huber's M-estimator, percentage of samples deemed uncorrupted, in
+        (0, 1].
+        This estimator is a trade-off between Tyler's estimator (q=0) and the
+        sample covariance matrix (q=1).
+    nu : float, default=5
+        Using Student-t's M-estimator, degree of freedom for t-distribution
+        (strictly positive).
+        This estimator is a trade-off between Tyler's estimator (nu->0) and the
+        sample covariance matrix (nu->inf).
 
     Returns
     -------
     cov : ndarray, shape (n_channels, n_channels)
-        Fixed point covariance matrix.
+        Robust M-estimator based covariance matrix.
 
     Notes
     -----
     .. versionadded:: 0.3.1
 
     References
     ----------
-    .. [1] `A distribution-free M-estimator of multivariate scatter
+    .. [1] `Complex Elliptically Symmetric Distributions: Survey, New Results
+        and Applications
+        <https://www.researchgate.net/profile/H-Vincent-Poor/publication/258658018_Complex_Elliptically_Symmetric_Distributions_Survey_New_Results_and_Applications/links/550480100cf24cee3a0150e2/Complex-Elliptically-Symmetric-Distributions-Survey-New-Results-and-Applications.pdf>`_
+        E. Ollila, D.E. Tyler, V. Koivunen, H.V. Poor. IEEE Transactions on
+        Signal Processing, 2012.
+    .. [2] `Robust antenna array processing using M-estimators of
+        pseudo-covariance
+        <http://lib.tkk.fi/Diss/2010/isbn9789526030319/article5.pdf>`_
+        E. Ollila, V. Koivunen. PIMRC, 2003.
+    .. [3] `Influence functions for array covariance matrix estimators
+        <https://ieeexplore.ieee.org/abstract/document/1289447/>`_
+        E. Ollila, V. Koivunen. IEEE SSP, 2003.
+    .. [4] `A distribution-free M-estimator of multivariate scatter
         <https://projecteuclid.org/journals/annals-of-statistics/volume-15/issue-1/A-Distribution-Free-M-Estimator-of-Multivariate-Scatter/10.1214/aos/1176350263.full>`_
-        D. E. Tyler. The Annals of Statistics, 1987.
-    .. [2] `Theoretical analysis of an improved covariance matrix estimator in
-        non-Gaussian noise
-        <https://hal.science/hal-02495012/document>`_
-        F. Pascal, P. Forster, J.P. Ovarlez, P. Arzabal. IEEE ICASSP, 2005.
-    .. [3] `Covariance structure maximum-likelihood estimates in compound
-        Gaussian noise: Existence and algorithm analysis
-        <https://hal.science/hal-01816367/document>`_
-        F. Pascal, Y. Chitour, J.P. Ovarlez, P. Forster, P. Arzabal. IEEE
-        Transactions on Signal Processing, 2008.
+        D.E. Tyler. The Annals of Statistics, 1987.
     """  # noqa
     n_channels, n_times = X.shape
+
+    if m_estimator == 'tyl':
```

Comment:
Other M-estimators also exist:
<img width="1063" alt="image" src="https://user-images.githubusercontent.com/44167796/218408059-88677d25-b11f-437b-be7f-8e37975816f3.png">

Extract from Gordana Draskovich's Thesis which isn't public but which I can send you privately.

Reply:
Thx for this nice summary table!
Other estimators can be added later, if required by pyRiemann users.